packages/functional-tests/tests/react-conversion/signup.spec.ts

```diff
@@ -3,96 +3,122 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { FirefoxCommand, createCustomEventDetail } from '../../lib/channels';
-import { expect, test } from '../../lib/fixtures/standard';
+import { expect, test as base } from '../../lib/fixtures/standard';
 import {
   syncDesktopV3QueryParams,
   syncMobileOAuthQueryParams,
 } from '../../lib/query-params';
 
+const AGE_12 = '12';
+const AGE_21 = '21';
 const PASSWORD = 'passwordzxcv';
 
-let email;
-
 const eventDetailLinkAccount = createCustomEventDetail(
   FirefoxCommand.LinkAccount,
   {
     ok: true,
   }
 );
 
-test.beforeEach(async ({ pages: { configPage, login } }) => {
+export const test = base.extend({
+  email: async ({ target, pages: { login } }, use) => {
+    const email = login.createEmail('signup_react{id}');
+    await login.clearCache();
+
+    await use(email);
+
+    try {
+      const creds = await target.auth.signIn(email, PASSWORD);
+      await target.auth.accountDestroy(email, PASSWORD, {}, creds.sessionToken);
+    } catch (error) {
+      // If the account is not created during the execution of the test, which
+      // can happen if the test fails prematurly, an exception will be thrown
+      // during email cleanup
+      const ERROR = 'Bad Request';
+      const MESSAGE = 'Unknown account';
+      if (!(error.error === ERROR && error.message == MESSAGE)) {
+        throw error;
+      }
```

Comment:
I noticed in some other spots, we're calling `await target.auth.accountDestroy` without a `try`/`catch`. Do we want to handle those cases similarly (in which case maybe we want a handler function we can pass the error to)? And if not, why do we want this check for specifically this suite?

Reply:
In other spots the try/catch should be present or better yet the `target.auth.accountDestroy` could have a check to make sure an account exists before it destroys it. In this test module in particular we have 'unhappy' paths where accounts don't get created or if tests fail early the accounts don't get created. In both these cases an error is thrown by `target.auth.accountDestroy`, which gives a false negative or can suppress/obfuscate the original error causing test failure. I think before this test module had a catch-all statement, I did make it specific, but for sure don't think this solution is ideal.

Ankita is currently working on [FXA-8984](https://mozilla-hub.atlassian.net/browse/FXA-8984), which i think will serve to make the behavior centralized and consistent. I hesitate to do more than refactor the before/after code into a fixture, as I've done here, as it's probably a duplication.